ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/InboundGovernor.hs

```diff
@@ -0,0 +1,109 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- Auxiliary module that gets stuff from 'Test.Ouroboros.Network.Server2'
+-- and adapts them in order to be used in Diffusion Tests.
+module Test.Ouroboros.Network.Testnet.Simulation.InboundGovernor where
+
+import           Ouroboros.Network.ConnectionManager.Types
+                     (Transition' (..))
+import           Ouroboros.Network.InboundGovernor
+                     (RemoteTransition, RemoteSt (..))
+
+import           Test.QuickCheck
+                     (Testable (property), counterexample)
+import           Test.Ouroboros.Network.Testnet.Simulation.ConnectionManager
+                     (AllProperty(..))
+
+
+-- | Pattern synonym which matches either 'RemoteHotEst' or 'RemoteWarmSt'.
+--
+pattern RemoteEstSt :: RemoteSt
+pattern RemoteEstSt <- (\ case
+                            RemoteHotSt  -> True
+                            RemoteWarmSt -> True
+                            _            -> False -> True
+                        )
+
+-- | Specification of the transition table of the inbound governor.
+--
+verifyRemoteTransition :: RemoteTransition -> Bool
```

Comment:
As in review of #3629, this function should be imported from a testing library in `ouroboros-network-framework`.

Reply:
Done!